cardano-cli/src/Cardano/CLI/Byron/Key.hs

```diff
@@ -58,13 +59,13 @@ renderByronKeyFailure err =
     VerificationKeyDeserialisationFailed vKeyFp deSerError ->
       "Error deserialising verification key at: " <> textShow vKeyFp <> " Error: " <> textShow deSerError
 
-newtype NewSigningKeyFile =
-  NewSigningKeyFile FilePath
-  deriving (Eq, Ord, Show, IsString)
```

Comment:
Note that we already do wrapping and unwrap here.

I'm thinking we can avoid this by eventually adding a type parameter to `File` like this:

```
newtype File direction content = File { unFile :: FilePath }
```

Then we can define:

```
data SigningKey
```

Which allows us to use `File In SigningKey` as the type of this file and no wrapping/unwrapping is necessary?

I haven't yet worked through the implications of this though.